File: CadastroCliente-App/src/app/_models/Cliente.ts
import { Endereco } from './Endereco';
import { Telefone } from './Telefone';

export class Cliente {
    clienteId: string;
    nome: string;
    documento: number;
    dataNascimento: Date;
    genero: string;
    endereco: Endereco;
    telefone: Telefone;
}

File: CadastroCliente.Business/Models/Cliente.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CadastroCliente.Business.Models
{
   public class Cliente
    {
        public Guid? ClienteId { get; set; }
        public string Nome { get; set; }
        public int Documento { get; set; }
        public DateTime DataNascimento { get; set; }
        public string Genero { get; set; }
        public Endereco Endereco { get; set; }
        public Telefone Telefone { get; set; }
    }
}
